js/utils.js

```diff
@@ -2102,6 +2110,50 @@ Fliplet.Registry.set('dynamicListUtils', (function() {
     });
   }
 
+  /**
+   * @param {Object} linkAction - link provider result object with required field `page`
+   * @param {String} queryParams - default query parameters that we want to add when navigating to another screen
+   *
+   * @returns {Promise} returns a promise of the Fliplet.Navigate.to method.
+   */
+  function navigateToTheScreen(linkAction, queryParams) {
+    if (!_.get(linkAction, 'page')) {
+      return Promise.reject('Page error');
+    }
+
+    linkAction.query = appendUrlQuery(linkAction.query, queryParams);
+
+    return Fliplet.Navigate.to(linkAction);
```

Comment:
The old function used a `try..catch` clause to capture sync errors (because `Fliplet.Navigate.to()` doesn't always return a promise).

![image](https://user-images.githubusercontent.com/290733/119686124-eccbe500-be3d-11eb-97ba-84504dde3849.png)

The new function doesn't seem to be using the same `try..catch` clause.